src/porepy/utils/interpolation_tables.py

```diff
@@ -345,32 +428,351 @@ def diff(self, x, axis):
             np.ndarray: Function values.
 
         """
-        # Fill any missing function values.
-        self._fill_values(x)
+        # Fill any missing function values if a function is available.
+        # NOTE: If the table is populated with the method 'assign_values', the user is
+        # responsible for assigning all necessary values.
+        if self._function is not None:
+            self._fill_values(x)
+
         # Use standard method for differentiation.
+        # NOTE: See comments in self.interpolation() regarding possible sources of
+        # unexpected errors from this function, and how to circumwent it.
         return super().diff(x, axis)
 
-    def _fill_values(self, x: np.ndarray):
-        # Find points in the interpolation grid that will be used for function
-        # evaluation. Compute function values as needed.
+    def assign_values(
+        self,
+        val: np.ndarray,
+        coord: np.ndarray,
+        indices: Optional[np.ndarray] = None,
+    ) -> None:
+        """Assign externally computed values to the table.
+
+        The user is responsible that the coordinates are nodes in the Cartesian grid
+        underlying this interpolation tabel.
+
+        This method can be used, e.g., if the table values are calculated by an
+        external function which it is not desirable to pass to the interpolation table.
+        One use case is when the function is an external library (e.g., a thermodynamic
+        flash calculation) which computes several properties at the same time. In this
+        case it is better to collect all properties externally and pass it to the
+        respective interpolation tables for the individual properties.
+
+        If the indices in the underlying Cartesian grid corresponding to the coordinate
+        points are known (e.g., the points were found by the method
+        interpolation_nodes_from_coordinates), it is strongly recommended that these are
+        also provided in assignment.
+
+        Args:
+            val: Values to assign.
+            coord: Coordinates of points to assign values to.
+            indices: Indices of the coordinates in the underlying Grid.
+
+        """
+
+        if indices is None:
+            # Get the corresponding indices in the underlying Cartesian grids.
+            inds = self._find_base_vertex(coord)
+        else:
+            inds = indices
+
+        ind_list = [i for i in inds.T]
+
+        # Add values and indices to the table.
+        column_permutation = self._table.add(ind_list, val, additive=False)
+
+        # Add the coordinates to the table. The table may have permuted the node
+        # coordinates during unification of the indices, the coordinates should be
+        # likewise permuted.
+        self._pt = np.hstack((self._pt, coord[:, column_permutation]))
+
+    def interpolation_nodes_from_coordinates(
```

Comment:
Consider renaming `interpolation_nodes_from_coordinates ` to something like `quadrature_points_from_coordinates`

Reply:
Agree. Will do.

Reply:
I will replace interpolation -> quadrature where natural in the documentation.